.github/workflows/run_tests.yml

```diff
@@ -9,6 +9,16 @@ on:
     - master
 
 jobs:
+
+  linting:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-python@v3
+      with:
+        python-version: '3.10'
+    - uses: pre-commit/action@v3.0.0
```

Comment:
Not an actionable item for @connortann , but @dsgibbons you might want to consider using pre-commit-ci rather than invoking via github actions in the future.

There's one potential limitation that I encountered before, which is the pre-commit-ci has a 3min timeout, whereas github actions is 1 hour(?). It is a moot point now, since ruff runs at lightning speed, but just FYI.

Reply:
Thank you for the suggestion @thatlittleboy - three minutes should be plenty. I've installed `pre-commit-ci`. Badge link here if you want it for `README.md` @connortann: [![pre-commit.ci status](https://results.pre-commit.ci/badge/github/dsgibbons/shap/master.svg)](https://results.pre-commit.ci/latest/github/dsgibbons/shap/master)

Reply:
I've taken your suggestion and removed the github action, so hopefully this PR is ready now.

Out of interest, is there any particular advantage to using precommit.ci rather than GitHub Actions? The repo already uses Actions to run the unit tests, so it seems clean to keep all checks together in one place and configured with code.

Reply:
Yep, 
1. pre-commit.ci autofixes in the PRs.
2. pre-commit-ci periodically runs autoupdate (so we don't need to manually update the versions inside `.pre-commit-config.yaml`) and raises a PR.
3. In theory, pre-commit-ci caches the hooks and versions used across all of its users, so we use `ruff, v0.0.256` and so did some other repo like `pandas-dev/pandas`, then it reuses that cache. So less wastage of resources, and probably runs faster also than GH actions, since it has less boilerplate to setup.